internal/pkg/manifest/validate.go

```diff
@@ -509,34 +509,23 @@ func (t TaskConfig) Validate() error {
 
 // Validate returns nil if PlatformArgsOrString is configured correctly.
 func (p PlatformArgsOrString) Validate() error {
-	if p.PlatformString != nil {
-		if err := p.PlatformString.Validate(); err != nil {
-			return err
+	if !p.PlatformArgs.isEmpty() {
+		if !p.PlatformArgs.bothSpecified() {
+			return errors.New(`fields "osfamily" and "architecture" must either both be specified or both be empty`)
 		}
+		return p.PlatformArgs.Validate()
 	}
-	return p.PlatformArgs.Validate()
-}
-
-// Validate returns nil if PlatformString is configured correctly.
-func (p PlatformString) Validate() error {
-	if err := validatePlatform(&p); err != nil {
-		return err
+	if p.PlatformString != nil {
+		if err := validateShortPlatform(p.PlatformString); err != nil {
+			return fmt.Errorf("validate platform: %w", err)
```

Comment:
We probably do not need to wrap this error, since `TaskConfig`'s `Validate` is wrapping this error with `validate "platform"` at line 495.

```suggestion
			return err
```